File: unity/Assets/Scripts/Multiplayer/Matchmaking/MatchmakingPlayersResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Response containing the list of players in a matchmaking lobby.
    /// </summary>
    /// <typeparam name="T">The type to deserialize player data into.</typeparam>
    [System.Serializable]
    public class MatchmakingPlayersResponse<T> : ApiResponse<EMatchmakingPlayersError> where T : class, new()
    {
        /// <summary>
        /// List of players currently in the matchmaking lobby.
        /// </summary>
        public List<MatchmakingPlayer<T>> players = new();

        /// <summary>
        /// Timestamp of when the player list was last updated.
        /// </summary>
        public string last_updated;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingPlayersResponse.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing the list of players in a matchmaking lobby.
 *
 * @param <T> The type to deserialize player data into.
 */
public class MatchmakingPlayersResponse<T> extends ApiResponse {
    @JsonProperty("players")
    private List<MatchmakingPlayer<T>> players;

    @JsonProperty("last_updated")
    private String lastUpdated;

    public List<MatchmakingPlayer<T>> getPlayers() {
        return players;
    }

    public void setPlayers(List<MatchmakingPlayer<T>> players) {
        this.players = players;
    }

    public String getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(String lastUpdated) {
        this.lastUpdated = lastUpdated;
    }
}
